### DISCUSSION TITLE
Vulnerable java version 17.0.1+12-39 not detected in docker image

### DISCUSSION THREAD

**Discussion by matjawor** [aquasecurity/trivy#5614] [Q&A] [answered]
### IDs

CVE-2022-34169

### Description

Trivy Version 0.47.0 scanned an image based on Rocky Linux 8.5 and java openjdk in version below:
```
openjdk 17.0.1 2021-10-19
OpenJDK Runtime Environment (build 17.0.1+12-39)
```
Trivy scan report does not includes an above-mentioned vulnerability. Java package was not installed via dnf package installer but manually prepared i.e.:

```bash
1. java openjdk downloaded and extracted
2. JAVA_HOME and PATH adjusted to openjdk usage
3. java and javac linked with a proper destination of executables through alternative command.
```

### Reproduction Steps

```bash
1. Build image based on Rocky Linux 8.5 and java openjdk in above-mentioned version
2. Run trivy image scanning
3. Get familiar with a report. Above-mentioned vulnerability will not be detected
...
```

### Target

Container Image

### Scanner

Vulnerability

### Target OS

Rocky Linux 8.5

### Debug Output

```bash
Inluded in the Description section.
```

### Version

```bash
0.47.0
```

### Checklist

- [x] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [x] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by DmitriyLewen** [comment, marked as answer]
Hello @matjawor 
Thanks for your report!

You installed openjdk manually. 
Therefore Trivy checks all extracted files as [language specific files](https://aquasecurity.github.io/trivy/v0.47/docs/scanner/vulnerability/#language-specific-packages).

About [CVE-2022-34169](https://github.com/advisories/GHSA-9339-86wc-4qgf):
This is java vulnerability.
But looks like openjdk doesn't contains `jar` file with ` xalan:xalan` package.

My check:
```
➜ tar -xf openjdk-17.0.2_linux-x64_bin.tar.gz 

➜ cd jdk-17.0.2 

➜ find . -name "*.jar"
./lib/jrt-fs.jar

➜ unzip -d jrt-fs ./lib/jrt-fs.jar 
...

➜ cat ./jrt-fs/META-INF/MANIFEST.MF 
Manifest-Version: 1.0
Specification-Title: Java Platform API Specification
Specification-Version: 17
Specification-Vendor: Oracle Corporation
Implementation-Title: Java Runtime Environment
Implementation-Version: 17.0.2
Implementation-Vendor: Oracle Corporation
Created-By: 16 (Oracle Corporation)

```

Regards, Dmitriy
